src/components/Api.js

```diff
@@ -0,0 +1,216 @@
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .then((res) => {
+        console.log(res);
+        this.funcObj.cardRequest(res);
+        this.funcObj.renderSavingNewCardForm();
+      })
+      .catch((err) => {
+        console.log("this is postCard catch");
+        console.error(err);
+      });
+  }
+
+  deleteCard(data) {
+    fetch(`${this.baseUrl}/cards/${data._id}`, {
+      method: "DELETE",
+      headers: {
+        authorization: this.token,
+        "content-type": "application/json",
+      },
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  likeCard(data) {
+    if (data.isLiked) {
+      fetch(`${this.baseUrl}/cards/${data._id}/likes`, {
+        method: "DELETE",
+        headers: {
+          authorization: this.token,
+          "content-type": "application/json",
+        },
+      })
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          }
+          return Promise.reject(`Error: ${res.status}`);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      fetch(`${this.baseUrl}/cards/${data._id}/likes`, {
+        method: "PUT",
+        headers: {
+          authorization: this.token,
+          "content-type": "application/json",
+        },
+      })
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          }
+          return Promise.reject(`Error: ${res.status}`);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  }
+
+  patchAvatar(data) {
+    fetch(`${this.baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: {
+        authorization: this.token,
+        "content-type": "application/json",
+      },
+      body: JSON.stringify({
+        avatar: data.url,
+      }),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .then((res) => {
+        this.funcObj.setAvatar({ url: res.avatar });
+        this.funcObj.renderSavingAvatarForm();
+      })
+      .catch((err) => {
+        console.log(err);
+        console.error(err);
+      });
+  }
+
+  getAvatar() {
+    fetch(`${this.baseUrl}/users/me`, {
+      headers: {
+        authorization: this.token,
+        "content-type": "application/json",
+      },
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .then((res) => {
+        this.funcObj.setAvatar({ url: res.avatar });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  getServerData() {
+    this.getUserData();
+    this.getAvatar();
+    this.getInitaialCards();
```

Comment:
You don't need to call `this.getAvatar();`  because when you call `this.getUserData();`, you get the avatar as well there